## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 426211`
Commit: `3dcde7ce7b7e47636c8041d57efa81ce69bfe6fe`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>httplz</li>
    <li>unbook</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 102 packages built:</summary>
  <ul>
    <li>actionlint</li>
    <li>alttab</li>
    <li>arcan</li>
    <li>arcan-all-wrapped</li>
    <li>arcan-wrapped</li>
    <li>arcan.dev</li>
    <li>arcan.lib</li>
    <li>arcan.man</li>
    <li>asn</li>
    <li>astroid</li>
    <li>atinout</li>
    <li>automirror</li>
    <li>bash-preexec</li>
    <li>bats</li>
    <li>blogc</li>
    <li>bupstash</li>
    <li>cached-nix-shell</li>
    <li>calibre</li>
    <li>cargo-update</li>
    <li>cat9-wrapped</li>
    <li>chickenPackages_5.chickenEggs.espeak</li>
    <li>cpcfs</li>
    <li>durden-wrapped</li>
    <li>dwarfs</li>
    <li>ekho</li>
    <li>emacspeak</li>
    <li>enroot</li>
    <li>espeak (espeak-ng)</li>
    <li>espeakup</li>
    <li>evil-winrm</li>
    <li>flock</li>
    <li>focus-stack</li>
    <li>gearlever</li>
    <li>geteltorito</li>
    <li>ghi</li>
    <li>git-identity</li>
    <li>google-guest-agent</li>
    <li>google-guest-configs</li>
    <li>gruut (python313Packages.gruut)</li>
    <li>gruut-ipa (python313Packages.gruut-ipa)</li>
    <li>gruut-ipa.dist (python313Packages.gruut-ipa.dist)</li>
    <li>gruut.dist (python313Packages.gruut.dist)</li>
    <li>ipcalc</li>
    <li>ipp-usb</li>
    <li>jogger</li>
    <li>kicadAddons.kikit</li>
    <li>kicadAddons.kikit-library</li>
    <li>kikit</li>
    <li>kikit.dist</li>
    <li>klavaro</li>
    <li>linux-router</li>
    <li>linux-router-without-wifi</li>
    <li>linux-wifi-hotspot</li>
    <li>locate-dominating-file</li>
    <li>mob</li>
    <li>ocserv</li>
    <li>packcc</li>
    <li>phonemizer (python313Packages.phonemizer)</li>
    <li>phonemizer.dist (python313Packages.phonemizer.dist)</li>
    <li>piper-phonemize</li>
    <li>piper-tts</li>
    <li>pipeworld-wrapped</li>
    <li>prio-wrapped</li>
    <li>python312Packages.espeak-phonemizer</li>
    <li>python312Packages.espeak-phonemizer.dist</li>
    <li>python312Packages.gruut</li>
    <li>python312Packages.gruut-ipa</li>
    <li>python312Packages.gruut-ipa.dist</li>
    <li>python312Packages.gruut.dist</li>
    <li>python312Packages.kokoro</li>
    <li>python312Packages.kokoro.dist</li>
    <li>python312Packages.misaki</li>
    <li>python312Packages.misaki.dist</li>
    <li>python312Packages.phonemizer</li>
    <li>python312Packages.phonemizer.dist</li>
    <li>python312Packages.piper-phonemize</li>
    <li>python312Packages.piper-phonemize.dist</li>
    <li>python313Packages.espeak-phonemizer</li>
    <li>python313Packages.espeak-phonemizer.dist</li>
    <li>python313Packages.kokoro</li>
    <li>python313Packages.kokoro.dist</li>
    <li>python313Packages.misaki</li>
    <li>python313Packages.misaki.dist</li>
    <li>python313Packages.piper-phonemize</li>
    <li>python313Packages.piper-phonemize.dist</li>
    <li>ronn</li>
    <li>roundup</li>
    <li>sbomnix</li>
    <li>sbomnix.dist</li>
    <li>speechd</li>
    <li>sycl-info</li>
    <li>termimage</li>
    <li>vulnix</li>
    <li>vulnix.dist</li>
    <li>vulnix.doc</li>
    <li>vulnix.man</li>
    <li>windmill</li>
    <li>wordbook</li>
    <li>xarcan</li>
    <li>zerotierone</li>
    <li>zerotierone.man</li>
    <li>zram-generator</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>httplz</summary>
<pre>   |
71 | unsafe impl Trait for ::std::marker::Send + Sync { }
   | ------------------------------------------------ first implementation here
72 | unsafe impl Trait for ::std::marker::Send + Send + Sync { }
73 | unsafe impl Trait for ::std::marker::Sync + Send { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(dyn Send + Sync + 'static)`

error[E0119]: conflicting implementations of trait `Trait` for type `(dyn Send + Sync + 'static)`
  --> /build/httplz-1.13.2-vendor/traitobject-0.1.0/src/impls.rs:75:1
   |
71 | unsafe impl Trait for ::std::marker::Send + Sync { }
   | ------------------------------------------------ first implementation here
...
75 | unsafe impl Trait for ::std::marker::Sync + Send + Sync { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(dyn Send + Sync + 'static)`

   Compiling bitflags v1.3.2
For more information about this error, try `rustc --explain E0119`.
error: could not compile `traitobject` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

--- a/pkgs/by-name/ev/evil-winrm/package.nix
+++ b/pkgs/by-name/ev/evil-winrm/package.nix
@@ -6,7 +6,14 @@
   bundlerEnv,
   bundlerUpdateScript,
 }:
-
+let
+  rubyEnv = bundlerEnv {
+    name = "evil-winrm";
+    gemfile = ./Gemfile;
+    lockfile = ./Gemfile.lock;
+    gemset = ./gemset.nix;
+  };
+in
 stdenv.mkDerivation rec {
   pname = "evil-winrm";
   version = "3.5";
@@ -18,19 +25,12 @@ stdenv.mkDerivation rec {
     hash = "sha256-8Lyo7BgypzrHMEcbYlxo/XWwOtBqs2tczYnc3+XEbeA=";
   };
 
-  env = bundlerEnv {
-    name = pname;
-    gemfile = ./Gemfile;
-    lockfile = ./Gemfile.lock;
-    gemset = ./gemset.nix;
-  };
-
   nativeBuildInputs = [
     makeWrapper
   ];
 
   buildInputs = [
-    env.wrappedRuby
+    rubyEnv.wrappedRuby
   ];
 
   installPhase = ''

--- a/pkgs/by-name/gh/ghi/package.nix
+++ b/pkgs/by-name/gh/ghi/package.nix
@@ -6,29 +6,30 @@
   bundlerEnv,
   tree,
 }:
-
-stdenv.mkDerivation (finalAttrs: {
-  pname = "ghi";
+let
   version = "1.2.1";
 
   src = fetchFromGitHub {
     owner = "drazisil";
     repo = "ghi";
-    rev = "refs/tags/${finalAttrs.version}";
+    tag = version;
     hash = "sha256-3V1lxI4VhP0jC3VSWyNS327gOCKowbbLB6ae1idpFFI=";
   };
 
-  env = bundlerEnv {
+  rubyEnv = bundlerEnv {
     name = "ghi";
-
-    gemfile = "${finalAttrs.src}/Gemfile";
-    lockfile = "${finalAttrs.src}/Gemfile.lock";
+    gemfile = "${src}/Gemfile";
+    lockfile = "${src}/Gemfile.lock";
     gemset = ./gemset.nix;
   };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ghi";
+  inherit version src;
 
   nativeBuildInputs = [ makeWrapper ];
 
-  buildInputs = [ finalAttrs.env.wrappedRuby ];
+  buildInputs = [ rubyEnv.wrappedRuby ];
 
   installPhase = ''
     mkdir -p $out/bin

--- a/pkgs/by-name/ro/ronn/package.nix
+++ b/pkgs/by-name/ro/ronn/package.nix
@@ -7,15 +7,15 @@
   groff,
   callPackage,
 }:
-
-stdenv.mkDerivation rec {
-  pname = "ronn";
-  version = env.gems.ronn-ng.version;
-
-  env = bundlerEnv {
+let
+  rubyEnv = bundlerEnv {
     name = "ronn-gems";
     gemdir = ./.;
   };
+in
+stdenv.mkDerivation {
+  pname = "ronn";
+  version = rubyEnv.gems.ronn-ng.version;
 
   dontUnpack = true;
 
@@ -27,7 +27,7 @@ stdenv.mkDerivation rec {
     runHook preInstall
 
     mkdir -p $out/bin
-    makeWrapper ${env}/bin/ronn $out/bin/ronn \
+    makeWrapper ${rubyEnv}/bin/ronn $out/bin/ronn \
       --set PATH ${groff}/bin
 
     runHook postInstall
@@ -46,6 +46,6 @@ stdenv.mkDerivation rec {
       zimbatm
       nicknovitski
     ];
-    platforms = env.ruby.meta.platforms;
+    platforms = rubyEnv.ruby.meta.platforms;
   };
 }

